src/vision/status_quo/Alan_tries_to_debug_a_hang.md

```diff
@@ -0,0 +1,134 @@
+    at /home/alan/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.4.0/src/runtime/driver.rs:198
+#12 0x000055555557a2f7 in tokio::runtime::basic_scheduler::Inner<P>::block_on::{{closure}} (scheduler=0x7fffffffcdb8, 
+    context=0x7fffffffcaf0)
+    at /home/alan/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.4.0/src/runtime/basic_scheduler.rs:224
+#13 0x000055555557b1b4 in tokio::runtime::basic_scheduler::enter::{{closure}} ()
+    at /home/alan/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.4.0/src/runtime/basic_scheduler.rs:279
+#14 0x000055555558174a in tokio::macros::scoped_tls::ScopedKey<T>::set (
+    self=0x555555701af8 <tokio::runtime::basic_scheduler::CURRENT>, t=0x7fffffffcaf0, f=...)
+    at /home/alan/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.4.0/src/macros/scoped_tls.rs:61
+#15 0x000055555557b0b6 in tokio::runtime::basic_scheduler::enter (scheduler=0x7fffffffcdb8, f=...)
+    at /home/alan/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.4.0/src/runtime/basic_scheduler.rs:279
+#16 0x0000555555579d3b in tokio::runtime::basic_scheduler::Inner<P>::block_on (self=0x7fffffffcdb8, future=...)
+    at /home/alan/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.4.0/src/runtime/basic_scheduler.rs:185
+#17 0x000055555557a755 in tokio::runtime::basic_scheduler::InnerGuard<P>::block_on (self=0x7fffffffcdb8, future=...)
+    at /home/alan/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.4.0/src/runtime/basic_scheduler.rs:425
+#18 0x000055555557aa9c in tokio::runtime::basic_scheduler::BasicScheduler<P>::block_on (self=0x7fffffffd300, future=...)
+    at /home/alan/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.4.0/src/runtime/basic_scheduler.rs:145
+#19 0x0000555555582094 in tokio::runtime::Runtime::block_on (self=0x7fffffffd2f8, future=...)
+    at /home/alan/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.4.0/src/runtime/mod.rs:450
+#20 0x000055555557c22f in inventory_service::main () at /home/alan/code/inventory_service/src/main.rs:4
+```
+
+</details>
+
+Puzzled, the only line Alan even recognizes is the `main` entry point function for the service.
+"Maybe I need to look at another thread?" he thinks and then consults the cheatsheet for the appropriate command.
+`info threads` seems promising so he tries that:
+
+<details><summary>(gdb) info threads</summary>
+
+```
+(gdb) info threads
+  Id   Target Id                                          Frame 
+* 1    Thread 0x7ffff7c3b5c0 (LWP 1048) "inventory_servi" 0x00007ffff7d5e58a in epoll_wait (epfd=3, events=0x555555711340, 
+    maxevents=1024, timeout=49152) at ../sysdeps/unix/sysv/linux/epoll_wait.c:30
+```
+
+</details>
+
+Alan is now even more confused: "Where are my tasks?" he thinks.
+After looking through the cheatsheet and StackOverflow, he discovers there isn't a way to see what async tasks are waiting to be woken up in the debugger.
```

Comment:
It might be worth expanding this part of the story a bit. I think Alan would probably know that tasks != threads but he also probably don't have a complete understanding of how exactly they work. He may think that there is one thread that loops through all the tasks and tries to wake them up or something.

Reply:
Yes, I think this would be helpful. You might also want to clarify whether Alan was using a multi-threaded runtime.